nemo_aligner/utils/trainer_utils.py

```diff
@@ -21,12 +21,23 @@
 
 
 def compute_num_steps_per_epoch(
-    sampler: Union[MegatronPretrainingRandomSampler, MegatronPretrainingRandomBatchSampler]
+    sampler: Union[MegatronPretrainingRandomSampler, MegatronPretrainingRandomBatchSampler],
+    limit_train_batches: Union[int, float] = 1.0,
 ):
     if not sampler.drop_last:
         raise NotImplementedError("`drop_last=False` is not currently supported")
 
-    return sampler.total_samples // sampler.global_batch_size
+    num_steps_per_epoch = sampler.total_samples // sampler.global_batch_size
+
+    if limit_train_batches is None:
+        limit_train_batches = 1.0
+
+    if limit_train_batches >= 0 and isinstance(limit_train_batches, float):
+        return min(num_steps_per_epoch, int(sampler.total_samples * limit_train_batches) // sampler.global_batch_size)
+    elif limit_train_batches >= 0 and isinstance(limit_train_batches, int):
+        return min(num_steps_per_epoch, limit_train_batches)
+    else:
+        return num_steps_per_epoch
```

Comment:
why can't we use the compute limit batches function below?

Reply:
if you want to support the case of `-1` then we can edit the compute_limit_batches function for that

Reply:
It doesn't really work to use the other function because they're just different enough on what they operate on to not be useful without a bunch of extra logic. The problem is that when we apply in the float case, we want to do `int(sampler.total_samples * limit_train_batches) // sampler.global_batch_size` meaning we apply the float on total samples **before** we divide by GBS, but in the other function, there's no way to easily squeeze that in.

And in terms of precision, they aren't the same:

```
int( int(7791 // 256) * 0.71 ) != int(7791 * 0.71) // 256
```

Reply:
in the integer limit_train_batches case i think
```
compute_limit_batches(num_steps_per_epoch, limit_train_batches)
```
should be correct.

for the float case, i agree there is some precision loss but because we quantize it to an integer I'm not convinced it actually matters

in your above example

`int( int(7791 // 256) * 0.71 ) != int(7791 * 0.71) // 256` both become 21

Reply:
as we discussed offline: 5f3746b12f825a64758792e70844de66f903054a to use compute limit batches. once CI passes will merge